lib/node_modules/@stdlib/math/base/special/cinvf/test/fixtures/julia/runner.jl

```diff
@@ -0,0 +1,116 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import JSON
+
+"""
+    gen( re, im, name )
+
+Generate fixture data and write to file.
+
+# Arguments
+
+* `re`: real components for first complex number
+* `im`: imaginary components for first complex number
+* `name::AbstractString`: output filename
+
+# Examples
+
+``` julia
+julia> re = rand( 1000 );
+julia> im = rand( 1000 );
+julia> gen( re, im, \"data.json\" );
+```
+"""
+function gen( re, im, name )
+	zre = Array{Float32}( undef, length(re) );
+	zim = Array{Float32}( undef, length(re) );
+	for i in eachindex(re)
+		z = 1.0 / Complex{Float32}( re[i], im[i] );
+		zre[ i ] = real( z );
+		zim[ i ] = imag( z );
+	end
+
+	# Store data to be written to file as a collection:
+	data = Dict([
+		("re", re),
+		("im", im),
+		("qre", zre),
+		("qim", zim),
+	]);
+
+	# Based on the script directory, create an output filepath:
+	filepath = joinpath( dir, name );
+
+	# Write the data to the output filepath as JSON:
+	outfile = open( filepath, "w" );
+	write( outfile, JSON.json(data) );
+	write( outfile, "\n" );
+	close( outfile );
+end
+
+# Get the filename:
+file = @__FILE__;
+
+# Extract the directory in which this file resides:
+dir = dirname( file );
+
+# Large positive real components:
+re = rand( Float32, 500 ) .* 1.0e300;
+im = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+gen( re, im, "large_positive_real_components.json" );
+
+# Large negative real components:
+re = -rand( Float32, 500 ) .* 1.0e300;
+im = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+gen( re, im, "large_negative_real_components.json" );
+
+# Large positive imaginary components:
+re = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+im = rand( Float32, 500 ) .* 1.0e300;
+gen( re, im, "large_positive_imaginary_components.json" );
+
+# Large negative imaginary components:
+re = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+im = -rand( Float32, 500 ) .* 1.0e300;
+gen( re, im, "large_negative_imaginary_components.json" );
+
+# Tiny positive real components:
+re = rand( Float32, 500 ) .* 1.0e-324;
+im = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+gen( re, im, "tiny_positive_real_components.json" );
+
+# Tiny negative real components:
+re = -rand( Float32, 500 ) .* 1.0e-324;
+im = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+gen( re, im, "tiny_negative_real_components.json" );
+
+# Tiny positive imaginary components:
+re = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+im = rand( Float32, 500 ) .* 1.0e-324;
+gen( re, im, "tiny_positive_imaginary_components.json" );
+
+# Tiny negative imaginary components:
+re = ( rand( Float32, 500 ) .* 20.0 ) .- 10.0;
+im = -rand( Float32, 500 ) .* 1.0e-324;
+gen( re, im, "tiny_negative_imaginary_components.json" );
```

Comment:
how to determine which values to use here, i'd be happy to learn for future packages as well!

Reply:
Currently, you are using numbers such as `1.0e300`, whereas the maximum value represented by a single-precision floating point number is around [`1.0e38`](https://en.wikipedia.org/wiki/Single-precision_floating-point_format). You'll need to adjust these calculations, so that the result falls in the range of single-precision floating-point numbers.